ion/symboltable.go

```diff
@@ -37,6 +38,8 @@ type SymbolTable interface {
 	WriteTo(w Writer) error
 	// String returns an ion text representation of the symbol table.
 	String() string
+	// InternToken adds a new symbol to this symbol table, or finds an existing definition of it.
+	InternToken(symbol string) SymbolToken
```

Comment:
`SymbolTable`s are meant to be immutable. Could this go on `SymbolTableBuilder` instead?

Reply:
Removed this API because symboltablebuilder has `Add()`